File: tuesday/morning/claude_exercise/figure1.f90
program figure1
    implicit none
    integer, parameter :: n_points = 1000
    real(8) :: x, dx
    real(8), dimension(n_points) :: x_vals, g_vals, diagonal
    integer :: i

    ! Generate x values from 0 to 1
    dx = 1.0d0 / dble(n_points - 1)
    do i = 1, n_points
        x_vals(i) = dble(i - 1) * dx
        ! Calculate g(x) = 4x(1-x)
        g_vals(i) = 4.0d0 * x_vals(i) * (1.0d0 - x_vals(i))
        ! 45 degree line
        diagonal(i) = x_vals(i)
    end do

    ! Write to file
    open(unit=10, file='figure1_data.txt', status='replace')
    write(10, '(A)') '# x g(x) diagonal'
    do i = 1, n_points
        write(10, '(3F20.10)') x_vals(i), g_vals(i), diagonal(i)
    end do
    close(10)

    print *, 'Figure 1 data written to figure1_data.txt'

end program figure1

File: tuesday/morning/claude_exercise/figure1.c
#include <stdio.h>
#include <stdlib.h>

#define N_POINTS 1000

int main() {
    double x_vals[N_POINTS];
    double g_vals[N_POINTS];
    double diagonal[N_POINTS];
    double dx;
    int i;
    FILE *fp;

    // Generate x values from 0 to 1
    dx = 1.0 / (N_POINTS - 1);
    for (i = 0; i < N_POINTS; i++) {
        x_vals[i] = i * dx;
        // Calculate g(x) = 4x(1-x)
        g_vals[i] = 4.0 * x_vals[i] * (1.0 - x_vals[i]);
        // 45 degree line
        diagonal[i] = x_vals[i];
    }

    // Write to file
    fp = fopen("figure1_data_c.txt", "w");
    if (fp == NULL) {
        fprintf(stderr, "Error opening file\n");
        return 1;
    }

    fprintf(fp, "# x g(x) diagonal\n");
    for (i = 0; i < N_POINTS; i++) {
        fprintf(fp, "%.10f %.10f %.10f\n", x_vals[i], g_vals[i], diagonal[i]);
    }
    fclose(fp);

    printf("Figure 1 data written to figure1_data_c.txt\n");

    return 0;
}
